### PR TITLE
USM: new configuration path service_monitoring_config.enable_go_tls_support

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Moves gotls to be under service monitoring instead of networks

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by nplanel — APPROVED** [review]

File: pkg/metadata/inventories/README.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ The payload is a JSON dict with the following fields
     `network_config.enabled` config option in `system-probe.yaml`).
   - `feature_networks_http_enabled` - **bool**: True if HTTP monitoring is enabled for Network Performance Monitoring (see: `network_config.enable_http_monitoring` config option in `system-proble.yaml`).
   - `feature_networks_https_enabled` - **bool**: True if HTTPS monitoring is enabled for Network Performance Monitoring (see: `network_config.enable_https_monitoring` config option in `system-proble.yaml`).
-  - `feature_networks_gotls_enabled` - **bool**: True if HTTPS monitoring through GoTLS is enabled for Network Performance Monitoring (see: `system_probe_config.enable_go_tls_support` config option in `system-proble.yaml`).
+  - `feature_usm_go_tls_enabled` - **bool**: True if HTTPS monitoring through GoTLS is enabled for Universal Service Monitoring (see: `service_monitoring_config.enable_go_tls_support` config option in `system-proble.yaml`).
```
Comment (val06) [inline pkg/metadata/inventories/README.md:96]: system-probe.yaml (not system-prob**l**e.yaml)

**Review by val06 — CHANGES_REQUESTED** [review]
cred

File: pkg/metadata/inventories/README.md (hunk 1 above)
Comment (guyarb) [reply pkg/metadata/inventories/README.md]: fixed

**Review by val06 — APPROVED** [review]

**Review by Kaderinho — COMMENTED** [review]
Was this option GA'ed ? If it's the case, I think we should consider this as a breaking change and it might be worth to also keep the previous config as an alias, or add a release note to report this change. WDYT ?

**Review by Kaderinho — APPROVED** [review]

**Review by maycmlee — COMMENTED** [review]
small edit

**Review by maycmlee — APPROVED** [review]
👍 for docs

**Pull request merged**